src/main/java/com/microsoft/jenkins/azuread/GraphClientCache.java

```diff
@@ -35,14 +40,23 @@ public class GraphClientCache {
             .build(GraphClientCache::createGraphClient);
 
     private static GraphServiceClient<Request> createGraphClient(GraphClientCacheKey key) {
-        final ClientSecretCredential clientSecretCredential = getClientSecretCredential(key);
 
+        TokenCredentialAuthProvider authProvider;
         String graphResource = AzureEnvironment.getGraphResource(key.getAzureEnvironmentName());
 
-        final TokenCredentialAuthProvider authProvider = new TokenCredentialAuthProvider(
-                singletonList(graphResource + ".default"),
-                clientSecretCredential
-        );
+        if ("Secret".equals(key.getCredentialType())) {
+            ClientSecretCredential clientSecretCredential = getClientSecretCredential(key);
+            authProvider = new TokenCredentialAuthProvider(
```

Comment:
this block can be factored out to a method that returns a `TokenCredential` so that we don't duplicate the auth provider line (and we shorten this method by 10 lines or so

Reply:
Refactored, please review.

Reply:
Close I pushed another commit to tidy more